sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/ChangeFeedStateTest.java

```diff
@@ -5,22 +5,47 @@
 
 import com.azure.cosmos.implementation.changefeed.common.ChangeFeedMode;
 import com.azure.cosmos.implementation.changefeed.common.ChangeFeedStartFromInternal;
+import com.azure.cosmos.implementation.changefeed.common.ChangeFeedStartFromTypes;
 import com.azure.cosmos.implementation.changefeed.common.ChangeFeedState;
 import com.azure.cosmos.implementation.changefeed.common.ChangeFeedStateV1;
 import com.azure.cosmos.implementation.feedranges.FeedRangeContinuation;
 import com.azure.cosmos.implementation.feedranges.FeedRangePartitionKeyRangeImpl;
 import com.azure.cosmos.implementation.query.CompositeContinuationToken;
 import com.azure.cosmos.implementation.routing.Range;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
 import java.nio.charset.StandardCharsets;
+import java.time.Instant;
 import java.util.Base64;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Map;
 import java.util.UUID;
 
+import static com.azure.cosmos.implementation.TestUtils.mockDiagnosticsClientContext;
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class ChangeFeedStateTest {
+    @DataProvider(name = "populateRequestArgProvider")
+    public Object[][] populateRequestArgProvider() {
+        return new Object[][] {
+            // changeFeed mode, changeFeed startFrom type, use continuation
+            { ChangeFeedMode.INCREMENTAL, ChangeFeedStartFromTypes.BEGINNING, true },
+            { ChangeFeedMode.INCREMENTAL, ChangeFeedStartFromTypes.NOW, true },
+            { ChangeFeedMode.INCREMENTAL, ChangeFeedStartFromTypes.POINT_IN_TIME, true },
+            { ChangeFeedMode.INCREMENTAL, ChangeFeedStartFromTypes.BEGINNING, false },
+            { ChangeFeedMode.INCREMENTAL, ChangeFeedStartFromTypes.NOW, false },
+            { ChangeFeedMode.INCREMENTAL, ChangeFeedStartFromTypes.POINT_IN_TIME, false },
+            { ChangeFeedMode.FULL_FIDELITY, ChangeFeedStartFromTypes.BEGINNING, true },
+            { ChangeFeedMode.FULL_FIDELITY, ChangeFeedStartFromTypes.NOW, true },
+            { ChangeFeedMode.FULL_FIDELITY, ChangeFeedStartFromTypes.POINT_IN_TIME, true },
+            { ChangeFeedMode.FULL_FIDELITY, ChangeFeedStartFromTypes.BEGINNING, false },
+            { ChangeFeedMode.FULL_FIDELITY, ChangeFeedStartFromTypes.NOW, false },
+            { ChangeFeedMode.FULL_FIDELITY, ChangeFeedStartFromTypes.POINT_IN_TIME, false },
```

Comment:
Full fidelity doesn't support point in time.

Reply:
Are we testing only to create ChangeFeedState?

Reply:
With this change, we should also test through our live tests for FullFidelityChangeFeedTest .

Reply:
Thanks, removed the unapplicable settings. 

The testing for this one is a little bit tricky, as we need to examine the final headers we are going to sent to backend. I did local e2e test for this change to check the point in time info has always been sent through the header. Will need to think about about how to do header validation in our tests.

Reply:
And FullFidelityChangeFeedTest, it should not impact it much, as point in time is not applicable.